contracts/core/mpl-rewards/v1/test/helper/Rewards.sol

```diff
@@ -0,0 +1,740 @@
+            fez.approve(address(bPool),    address(stakeLocker1), amt);
+            assertTrue(fez.try_stakeTo(    address(stakeLocker1), amt));
+            pat.setStakeLockerLockupPeriod(address(stakeLocker1),    0);
+            assertEq(stakeToken.balanceOf(address(fez)), amt);
+        }
+
+        assertEq(stakeToken.custodyAllowance(address(fez), address(custodian)), 0);
+        assertEq(stakeToken.totalCustodyAllowance(address(fez)),                0);
+
+        fez.increaseCustodyAllowance(address(custodian), custodyAmt);
+
+        assertEq(stakeToken.custodyAllowance(address(fez), address(custodian)), custodyAmt);
+        assertEq(stakeToken.totalCustodyAllowance(address(fez)),                custodyAmt);
+
+        uint256 pullBackAmt = (amt - custodyAmt);
+
+        assertEq(stakeToken.balanceOf(address(fez)), amt);
+
+        if (isPfdtStakeAsset) {
+            make_withdrawable(fez, pool1);
+
+            assertTrue(!fez.try_withdraw(address(stakeToken), toUsd(pullBackAmt) + 1));
+            assertTrue( fez.try_withdraw(address(stakeToken),     toUsd(pullBackAmt)));
+            assertEq(  usdc.balanceOf(   address(fez)),             toUsd(pullBackAmt));
+
+        } else {
+            make_unstakeable(Staker(address(fez)), stakeLocker1);
+
+            assertTrue(!fez.try_unstake(address(stakeToken), pullBackAmt + 1));
+            assertTrue( fez.try_unstake(address(stakeToken),     pullBackAmt));
+        }
+
+        assertEq(stakeToken.balanceOf(address(fez)), custodyAmt);
+    }
+
+    function fdt_transferByCustodian(uint256 amt, uint256 custodyAmt, bool isPfdtStakeAsset, IStakeToken stakeToken) public {
+        Custodian custodian = new Custodian();  // Custodial contract for FDTs - will start out as liquidity mining but could be broader DeFi eventually
+
+        if (isPfdtStakeAsset) {
+            amt         = constrictToRange(amt,        1, 1E9, true);  // $1 - $1b
+            custodyAmt  = constrictToRange(custodyAmt, 1, amt, true);  // $1 - deposit
+
+            mintFundsAndDepositIntoPool(fay, pool1, amt * USD, amt * USD);
+
+            amt  *= WAD;
+            custodyAmt  *= WAD;
+        } else {
+            // Transfer BPTs to the Farmers i.e Fay and Fez. Although both can use interchangeably but to make it code consistent chosses to use Fay anf Fez. 
+            sam.transfer(address(bPool), address(fay), bPool.balanceOf(address(sam)));
+
+            amt        = constrictToRange(amt,        1, bPool.balanceOf(address(fay)), true);  // 1 wei - whole BPT bal
+            custodyAmt = constrictToRange(custodyAmt, 1, amt,                           true);  // 1 wei - amt
+
+            // Make StakeLocker public and stake tokens
+            pat.openStakeLockerToPublic(        address(stakeLocker1));
+            fay.approve(address(bPool),         address(stakeLocker1), amt);
+            assertTrue(fay.try_stakeTo(         address(stakeLocker1), amt));
+        }
```

Comment:
it is not always the same so I did not extract an internal function out of it but if we are going to have more generic functions as same then maybe in the future it will be good to extract one out of it.